RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -0,0 +1,416 @@
+
+		switch state.mode {
+		case let .factorSourceWithID(_, loadableState):
+			switch loadableState {
+			case let .success(factorSource):
+				factorSourceOption = .specific(factorSource)
+			default:
+				let errorMsg = "Discrepancy! Expected to loaded the factor source"
+				loggerGlobal.error(.init(stringLiteral: errorMsg))
+				assertionFailure(errorMsg)
+				return .none
+			}
+		case let .privateHD(privateHDFactorSource):
+			factorSourceOption = .specificPrivateHDFactorSource(privateHDFactorSource)
+		}
+
+		state.status = .derivingPublicKeys
+		state.destination = .derivePublicKeys(.init(
+			derivationPathOption: .knownPaths(
+				Array(derivationPaths),
+				networkID: state.networkID
+			),
+			factorSourceOption: factorSourceOption,
+			purpose: .createEntity(kind: .account)
+		))
+
+		return .none
+	}
+
+	private func scanOnLedger(
+		accounts: IdentifiedArrayOf<Profile.Network.Account>,
+		state: inout State
+	) -> Effect<Action> {
+		assert(accounts.count == batchSize)
+		state.status = .scanningNetworkForActiveAccounts
+		state.destination = nil
+		return .run { send in
+			let (active, inactive) = try await performScan(accounts: accounts)
+			await send(.internal(.foundAccounts(active: active, inactive: inactive)))
+		}
+	}
+
+	private func performScan(
+		accounts: IdentifiedArrayOf<Profile.Network.Account>
+	) async throws -> (
+		active: IdentifiedArrayOf<Profile.Network.Account>,
+		inactive: IdentifiedArrayOf<Profile.Network.Account>
+	) {
+		let accountAddresses: [AccountAddress] = accounts.map(\.address)
+
+		do {
+			let activeAccounts = try await onLedgerEntitiesClient
+				.getAccounts(
+					accountAddresses,
+					// actually we wanna `resourceMetadataKeys` as well here, but we cannot since
+					// the count will exceed `EntityMetadataKey.maxAllowedKeys`.
+					metadataKeys: [.ownerBadge, .ownerKeys],
+					forceRefresh: true
+				)
+				.filter { (onLedgerAccount: OnLedgerEntity.Account) -> Bool in
+
+					guard
+						case let metadata = onLedgerAccount.metadata,
+						let ownerKeys = metadata.ownerKeys,
+						let ownerBadge = metadata.ownerBadge
+					else { return false }
+
+					func hasStateChange(_ list: OnLedgerEntity.Metadata.ValueAtStateVersion<some Any>) -> Bool {
+						list.lastUpdatedAtStateVersion > 0
+					}
+					let isActive = hasStateChange(ownerKeys) || hasStateChange(ownerBadge)
+					return isActive
+				}
+
+			var active: IdentifiedArrayOf<Profile.Network.Account> = []
+			var inactive: IdentifiedArrayOf<Profile.Network.Account> = []
+			for account in accounts {
+				if let activeAccount = activeAccounts.first(where: { $0.address == account.address }) {
+					var account = account
+					if let depositRule = activeAccount.details?.depositRule {
+						account.onLedgerSettings.thirdPartyDeposits.depositRule = depositRule
+					}
+					active.append(account)
+				} else {
+					inactive.append(account)
+				}
+			}
+
+			// Don't remove this, this is in fact needed if we don't
```

Comment:
I don't have the full context, but why for this feature do we need to get also all of the related resources?

Reply:
hmm, depending on how this impacts the feature developed with this PR, we could have an API that "just" reads the info about the account from the ledger, and disregards any resources information.

Reply:
mm lets see if we address this before or after 1.3.0